### DISCUSSION TITLE
pymodbus road map

### DISCUSSION THREAD

**Discussion by MrWaloo** [pymodbus-dev/pymodbus#2317] [General]
Hi,

I just wanted to know if the road map of this project is documented somewhere. It could be helpful for people like me (absolutely no insider) not to answer annoying questions about functions that are meant to be outdated.

**Comment by janiversen** [comment]
We document all changes in the changelog and all api changes in a seperate document, that tells you clearly if a function is outdated.

We do not publish our roadmap, simply because it's highly dynamic. Do not forget that a lot of the commits comes from contributors that follow their own roadmap.

**Comment by MrWaloo** [comment]
> We document all changes in the changelog and all api changes in a seperate document that tells you clearly if a function is outdated.

Absolutely and the documentation is well written! As an non insider, I had to make my own tests for some functions, but it is fun to do and I could learn the API by this way.

So it is possible to see what has been done. My question was about what will be done.
For instance, I was not aware of the fact that BinaryPayload* classes will be removed in the next major version (3.8.0). I only know it since you told me. Now, with the access you granted me, I can see more branches and can check what is currently ongoing.

Except in the issues, the not yet committed PR and the branches, there is no todo list?

In the readme:
> There are 2 bigger projects ongoing:
> - rewriting the internal part of all clients (both sync and async)
> - Add features to and simulator, and enhance the web design

Without details, that's why I am asking. But it's absolutely OK if there is no "online" answer.

Thanks

**Reply by janiversen** [reply]
As I said the development are to a large extent driven by contributors, and the maintainers is working (slowly) on the 2 bigger projects, so there are no roadmap. The open issues gives you an idea of what is going to happen.

The binarypayload is a good example, it would be axed long term as a cleanup, but the discussion about bit patterns accelerated that.

Unless you want to help with the development, a roadmap really does not help you, since you will be using what is released not ideas for a potential future.

I did not grant you more access, but pushed my development branches for another purpose.....these branches do not indicate what will actually be released.
